src/library/trackcollectionmanager.cpp

```diff
@@ -221,26 +221,51 @@ TrackCollectionManager::SaveTrackResult TrackCollectionManager::saveTrack(
         return SaveTrackResult::Skipped;
     }
 
-    // The dirty flag is reset while saving the track in the internal
-    // collection!
+    if (!pTrack->getId().isValid()) {
+        // Track has been purged from the internal collection/database
+        // while it was cached in-memory.
+        // TODO: Is this race condition even possible?? The debug assertion
+        // in TrackDAO::saveTrack() never triggered so it must at least be
+        // very unlikely.
+        if (!m_externalCollections.isEmpty()) {
+            kLogger.debug()
+                    << "Purging deleted track"
+                    << pTrack->getLocation()
+                    << "from"
+                    << m_externalCollections.size()
+                    << "external collection(s)";
+            for (const auto& externalTrackCollection : qAsConst(m_externalCollections)) {
```

Comment:
```suggestion
            for (const auto& externalTrackCollection : std::as_const(m_externalCollections)) {
```
doesn't make a difference, but I prefer the standard library over Qt. opinions?

Reply:
I only restructured the code, no other changes.

Reply:
I tried to restrict the changes. Replacing a single occurrence doesn't make any sense to me. Now I replaced all occurrences in this file.

All these changes raise the bar for starting a review. That's why I avoided them in the first place.

Reply:
I'm sorry. I didn't mean to widen the scope of the PR. I was just looking at the code and saw an opportunity for improvement (see the [boy scout rule](https://www.oreilly.com/library/view/97-things-every/9780596809515/ch08.html). 
Thanks for taking care of the rest of the file too though.

Reply:
Yeah, I think it's okay to point stuff like this out, but it's also okay for the PR autbor to reject the change and keep the scope small. We can always do cleanups like that in a follow up PR.